### PR TITLE
Retrigger Geo areas in Project edit page

### PR DESCRIPTION
**Pull request by shreeyash07**
- Addresses #3005
- Depends on [fix/geo-area-failure-status-trigger](https://github.com/the-deep/server/pull/1537)

## Changes

* Add retrigger mutation in Region card
* Add retrigger button in region card
* Add polling for the refetchRegion due to background operation
* Add disable condition for publish area button

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx [hunk 1]
```diff
@@ -151,6 +151,7 @@ interface Props {
     value: PartialAdminLevel;
     isPublished?: boolean;
     adminLevelOptions?: AdminLevelType[];
+    onRegionPublishSuccess: () => void;
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx]: Rename this appropriately

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 2]
```diff
@@ -226,6 +239,48 @@ function RegionCard(props: Props) {
...
+                        { variant: 'error' },
+                    );
+                }
+
+                if (ok) {
+                    alert.show(
+                        'Selected region retrigger successfully',
+                        { variant: 'success' },
+                    );
+                    onRegionPublishSuccess();
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: rename this approriately

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 3]
```diff
@@ -497,20 +567,36 @@ function RegionCard(props: Props) {
...
+                        Are you sure you want to remove this geo area from the project?"
+                        showConfirmationInitially={false}
+                        disabled={navigationDisabled}
+                    >
+                        <IoTrashBinOutline />
+                    </QuickActionConfirmButton>
+                </>
             )}
         >
+            {region.status === 'FAILED' && (
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: Check if the region is already published. If the region is already published this check is not necessary as the user won't/shouldn't be able to retrigger.

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 4]
```diff
@@ -497,20 +567,36 @@ function RegionCard(props: Props) {
...
-                    message="Removing the geo area will remove all the tagged geo data under your project.
-                    The removal of tags cannot be undone.
-                    Are you sure you want to remove this geo area from the project?"
-                    showConfirmationInitially={false}
-                    disabled={navigationDisabled}
-                >
-                    <IoTrashBinOutline />
-                </QuickActionConfirmButton>
+                <>
+                    {region.status === 'FAILED' && (
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: Check if the region is already published. If the region is already published this check is not necessary as the user won't/shouldn't be able to retrigger.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Review by subinasr — APPROVED** [review]

**Pull request merged**